hedera-node/src/main/java/com/hedera/services/state/logic/RecordStreaming.java

```diff
@@ -75,9 +99,43 @@ public void run() {
 	}
 
 	public void stream(final RecordStreamObject rso) {
+		if (isInNewBlock(rso)) {
+			Hash computedHash = new Hash();
+			try {
+				computedHash = curRunningHashLeaf.getRunningHash().getFutureHash().get();
```

Comment:
It would be nice to add a method to `RunningHashLeaf` like `getLatestBlockHash()` that does this method chain internally.